src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || str.length <= charLimit) {
```

Comment:
As a long term fix, indeed we could look at a re-sequence, the immediate issue appears resolved.

The potential conflict mentioned above appeared after running a hard truncate then applying newline truncate. If the newline truncate is meant to preserve the height of a container, then running that first in sequence appears to reduce the greater number of characters first. After that, as your question mentions, running a word boundary, or even a hard character limit, would help narrow it down further. 

Hard truncate and word boundary slightly boil down to interpretation, the sequence that currently exists, from observation, appears geared toward preserving enough space for a link, which in a way, slightly feels like the responsibility of the parent not the filter itself.